File: app/src/main/java/com/mgaetan89/showsrage/helper/DateTimeHelper.java
package com.mgaetan89.showsrage.helper;

import android.content.Context;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.text.format.DateUtils;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public abstract class DateTimeHelper {
	public static CharSequence getRelativeDate(String dateTime, String format, long minResolution) {
		if (TextUtils.isEmpty(dateTime)) {
			return "N/A";
		}

		try {
			DateFormat formatter = new SimpleDateFormat(format, Locale.getDefault());
			Date date = formatter.parse(dateTime);

			return DateUtils.getRelativeTimeSpanString(date.getTime(), System.currentTimeMillis(), minResolution);
		} catch (ParseException exception) {
			return dateTime;
		}
	}

	@Nullable
	public static String getLocalizedTime(Context context, String dateTime, String format) {
		long timestamp = getTimestamp(dateTime, format);

		if (timestamp == 0L) {
			return null;
		}

		return DateUtils.formatDateTime(context, timestamp, DateUtils.FORMAT_SHOW_TIME);
	}

	private static long getTimestamp(String dateTime, String format) {
		// SickRage returns the air time in the american time format
		DateFormat dateFormat = new SimpleDateFormat(format, Locale.US);

		try {
			Date date = dateFormat.parse(dateTime);

			return date.getTime();
		} catch (ParseException exception) {
			return 0L;
		}
	}
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/helper/DateTimeHelper.kt
package com.mgaetan89.showsrage.helper

import android.content.Context
import android.text.format.DateUtils
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Locale

object DateTimeHelper {
	fun getRelativeDate(dateTime: String?, format: String, minResolution: Long): CharSequence? {
		if (dateTime.isNullOrEmpty()) {
			return "N/A"
		}

		try {
			val date = SimpleDateFormat(format, Locale.getDefault()).parse(dateTime)

			return DateUtils.getRelativeTimeSpanString(date.time, System.currentTimeMillis(), minResolution)
		} catch (exception: ParseException) {
			return dateTime!!
		}
	}

	fun getLocalizedTime(context: Context, dateTime: String, format: String): String? {
		val timestamp = this.getTimestamp(dateTime, format).takeIf { it > 0L } ?: return null

		return DateUtils.formatDateTime(context, timestamp, DateUtils.FORMAT_SHOW_TIME)
	}

	private fun getTimestamp(dateTime: String, format: String): Long {
		return try {
			// SickRage returns the air time in the american time format
			SimpleDateFormat(format, Locale.US).parse(dateTime).time
		} catch (exception: ParseException) {
			0L
		}
	}
}
